Snake/snake.py

```diff
@@ -111,38 +119,43 @@ def draw_end(points: int):
 
 
 # check if the snake collides with the fruit
-def check_collision():
-    if snake[0]["x"] == fruit["x"] and snake[0]["y"] == fruit["y"]:
+def check_collision(fruit: Position):
+    if any(map(lambda seg: seg == fruit, snake.body)):
         return True
     return False
 
 
 # check if the snake collides with itself
 def check_self_collision():
-    for segment in snake[1:]:
-        if segment["x"] == snake[0]["x"] and segment["y"] == snake[0]["y"]:
+    for segment in snake.body[1:]:
+        if snake.body[0] == segment:
             return True
     return False
 
 
 # randomize the fruit position
 def randomize_fruit():
-    fruit["x"] = random.randint(0, SQAURE_SIZE - 1)
-    fruit["y"] = random.randint(0, SQAURE_SIZE - 1)
-
-    if check_collision():
-        randomize_fruit()
+    """#! This Code is flawed due to the issue that all fields could be occupied!
+    A faster and more reliable way to find a empty position is to choose a random start
+    and checking the closest positions systematically. This ensures, that a
+    result will always be found deterministically in at most Square_Size^2 steps!
+    While the current Implementation might never find a free spot.
+    """
+    while True:
+        fpos = Position.random_pos(SQAURE_SIZE, SQAURE_SIZE)
+        if not check_collision(fpos):
+            print(fpos)
```

Comment:
Question: Is it necessary to display the newly determined position of the fruit on the console?